pkg/kubelet/api/v1alpha1/runtime/api.proto

```diff
@@ -0,0 +1,625 @@
+    // The hash of container config
+
+    // Variables for interactive containers, these have very specialized
+    // use-cases (e.g. debugging).
+    // TODO: Determine if we need to continue supporting these fields that are
+    // part of Kubernetes's Container Spec.
+    optional bool stdin = 13;
+    optional bool stdin_once = 14;
+    optional bool tty = 15;
+
+    // Linux contains configuration specific to Linux containers.
+    optional LinuxContainerConfig linux = 16;
+}
+
+// LinuxContainerResources specifies Linux specific configuration for
+// resources.
+// TODO: Consider using Resources from opencontainers/runtime-spec/specs-go
+// directly.
+message LinuxContainerResources {
+    // CPU CFS (Completely Fair Scheduler) period
+    optional int64 cpu_period = 1;
+    // CPU CFS (Completely Fair Scheduler) quota
+    optional int64 cpu_quota = 2;
+    // CPU shares (relative weight vs. other containers)
+    optional int64 cpu_shares = 3;
+    // Memory limit in bytes
+    optional int64 memory_limit_in_bytes = 4;
+    // OOMScoreAdj adjusts the oom-killer score.
+    optional int64 oom_score_adj = 5;
+}
+
+// SELinuxOption are the labels to be applied to the container.
+message SELinuxOption {
+    optional string user = 1;
+    optional string role = 2;
+    optional string type = 3;
+    optional string level = 4;
+}
+
+// Capability contains the container capabilities to add or drop 
+message Capability {
+    // List of capabilities to add.
+    repeated string add_capabilities = 1;
+    // List of capabilities to drop.
+    repeated string drop_capabilities = 2;
+}
+
+// LinuxContainerConfig contains platform-specific configuration for
+// Linux-based containers.
+message LinuxContainerConfig {
+    // Resources specification for the container.
+    optional LinuxContainerResources resources = 1;
+    // Capabilities to add or drop.
+    optional Capability capabilities = 2;
+    // Optional SELinux context to be applied.
+    optional SELinuxOption selinux_options = 3;
+}
+
+message CreateContainerRequest {
+    // The id of the PodSandbox
+    optional string pod_sandbox_id = 1;
+    // The config of the container
+    optional ContainerConfig config = 2;
+    // The config of the PodSandbox
+    optional PodSandboxConfig sandbox_config = 3;
+}
+
+message CreateContainerResponse {
+    // The id of the created container
+    optional string container_id = 1;
+}
+
+message StartContainerRequest {
+    // The id of the container
+    optional string container_id = 1;
+}
+
+message StartContainerResponse {}
+
+message StopContainerRequest {
+    // The id of the container
+    optional string container_id = 1;
+    // Timeout in seconds to stop the container
+    optional int64 timeout = 2;
+}
+
+message StopContainerResponse {}
+
+message RemoveContainerRequest {
+    // The id of the container
+    optional string container_id = 1;
+}
+
+message RemoveContainerResponse {}
+
+enum ContainerState {
+    CREATED = 0;
```

Comment:
cc @euank For rkt, there is a state before "created" (aka "prepared"), called preparing, which rkt is preparing the fs for the container. Maybe we should not list those containers so that they don't get gc'd accidently.
Or maybe we add "CREATING" state here?